pulser/tests/test_simulation.py

```diff
@@ -474,7 +501,7 @@ def test_config():
 
 def test_noise():
     sim2 = Simulation(
-        seq, sampling_rate=0.01, config=SimConfig(noise=("doppler"))
+        seq, sampling_rate=0.01, config=SimConfig(noise=("SPAM"), eta=0.4)
     )
     sim2.run()
     with pytest.raises(NotImplementedError, match="Cannot include"):
```

Comment:
The reason why it's done that way is because `self.basis_name` is defined on the first time `self._construct_hamiltonian()` is called, so that check can only happen afterwards. I don't like it either, but it is not as simple as moving the check upwards, unfortunately

Reply:
Ok, I patched this one